File: 0019/0019.c
struct ListNode {
    int val;
    struct ListNode *next;
};

struct ListNode* removeNthFromEnd(struct ListNode* head, int n){
    struct ListNode* dummy = (struct ListNode*)malloc(sizeof(struct ListNode));
    struct ListNode* left = dummy;
    struct ListNode* right = dummy;
    struct ListNode* prev = left;
    int distance = 0;
    dummy->next = head;

    while(right->next){
        prev = left;
        if (distance == n - 1){
            left = left->next;
        }
        else{
           distance ++; 
        }
        right = right->next;
    }

    prev->next = prev->next->next;
    
    return dummy->next;
}

File: 0019/0019.py
from typing import *

class ListNode:
    def __init__(self, val=0, next=None):
        self.val = val
        self.next = next

class Solution:
    def removeNthFromEnd(self, head: Optional[ListNode], n: int) -> Optional[ListNode]:
        dummy = ListNode(next=head)
        left, right, distance = dummy, dummy, 0
        while (right.next):
            prev = left
            if (distance == n - 1):
                left = left.next
            else:
                distance += 1    
            right = right.next

        prev.next = prev.next.next

        return dummy.next